manifests/init.pp

```diff
@@ -46,7 +46,10 @@
   $url,
   $destination_directory,
   $destination_file = '',
-  $proxyAddress=''
+  $proxyAddress='',
+  $proxyUser='',
+  $proxyPassword='',
+  $isPasswordSecure=true
```

Comment:
Yes, exactly - node data cannot be pre-populated. So, many if not most users will likely forego this feature.